pallets/gear-program/src/migrations/ed_locks.rs

```diff
@@ -44,22 +47,32 @@ const MIGRATE_FROM_VERSION: u16 = 7;
 const MIGRATE_TO_VERSION: u16 = 8;
 const ALLOWED_CURRENT_STORAGE_VERSION: u16 = 9;
 
+// Redefine this constant from the `pallet_gear` crate to avoid a cyclic dependency.
+const EXISTENTIAL_DEPOSIT_LOCK_ID: [u8; 8] = *b"glock/ed";
+
 pub(crate) type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
 pub(crate) type CurrencyOf<T> = <T as pallet_treasury::Config>::Currency;
 pub(crate) type BalanceOf<T> = <CurrencyOf<T> as Currency<AccountIdOf<T>>>::Balance;
 
-pub struct MigrateToV8<T: Config>(PhantomData<T>);
+pub struct SetLocksOnED<T: Config>(PhantomData<T>);
 
-impl<T: Config> OnRuntimeUpgrade for MigrateToV8<T>
+impl<T: Config> OnRuntimeUpgrade for SetLocksOnED<T>
 where
     T: pallet_treasury::Config + pallet_balances::Config,
     T::AccountId: Origin,
     T::Paymaster: Pay<Beneficiary = T::AccountId, AssetKind = (), Balance = BalanceOf<T>>,
+    CurrencyOf<T>: LockableCurrency<T::AccountId>,
 {
     fn on_runtime_upgrade() -> Weight {
         let onchain = Pallet::<T>::on_chain_storage_version();
         let existential_deposit = CurrencyOf::<T>::minimum_balance();
-        let transfer_weight = <T as pallet_balances::Config>::WeightInfo::transfer_allow_death();
+        // Benchmarked value
+        let set_lock_weight = Weight::from_parts(12_000_000, 4764)
```

Comment:
I wish there were such a method in pallet Balances WeightInfo type but `set_lock()` is not an extrinsic. I had to benchmark on my own but since it's just a one-time thing there's no need to keep it.